gateway/handler_success.go

```diff
@@ -76,6 +77,15 @@ func addVersionHeader(w http.ResponseWriter, r *http.Request, globalConf config.
 	}
 }
 
+func addTraceID(w http.ResponseWriter, r *http.Request, globalConf config.Config) {
```

Comment:
I'd change this function signature to receive if Opentelemetry is enabled/disabled and move the whole function into `internal/otel` pkg

Reply:
@tbuchaillot makes total sense to do it instead of sending the whole config object, good catch! 👌